docs/build-speed.md

```diff
@@ -0,0 +1,200 @@
+---
+id: build-speed
+title: Speeding up your Build
+---
+
+Building your React Native app could be **expensive** and take several minutes of developers time.
+This can be problematic as your project grows and generally in bigger organizations with multiple React Native developers.
+
+With [the New React Native Architecture](/docs/next/new-architecture-app-modules-android), this problem is becoming more critical
+as you might have to compile some native C++ code in your project with the Android NDK in addition to the native code already necessary for the iOS and Android platforms.
+
+To mitigate this performance hit, this page shares some suggestions on how to **improve your build time**.
+
+## Android
+
+### Build only one ABI
+
+When building your android app locally, you build all the 4 ABIs by default: `armeabi-v7a`, `arm64-v8a`, `x86` & `x86_64`.
+
+However, you probably don't need to build all of them if you're building locally and testing your emulator or on a physical device.
+
+This should reduce your build time by a **~75% factor**.
+
+If you're using the React Native CLI, you can use the `--active-arch-only` flag together with the `run-android` command.
+This flag will make sure the correct ABI is picked up from either the running emulator or the plugged in phone.
+To confirm that this approach is working fine, you'll see a message like `info Detected architectures arm64-v8a` on console.
+
+```
+$ yarn react-native run-android --active-arch-only
+
+[ ... ]
+info Running jetifier to migrate libraries to AndroidX. You can disable it using "--no-jetifier" flag.
+Jetifier found 1037 file(s) to forward-jetify. Using 32 workers...
+info JS server already running.
+info Detected architectures arm64-v8a
+info Installing the app...
+```
+
+This mechanism relies on the `reactNativeArchitectures` Gradle property.
+
+Therefore, if you're building directly with Gradle from the command line and without the CLI, you can specify the ABI you want to build as follows:
+
+```
+$ ./gradlew :app:assembleDebug -PreactNativeArchitectures=x86,x86_64
+```
+
+This can be useful if you wish to build your Android App on a CI and use a matrix to parallelize the build of the different architectures.
+
+If you wish, you can also override this value locally, using the `gradle.properties` file you have in the [top level folder](https://github.com/facebook/react-native/blob/19cf70266eb8ca151aa0cc46ac4c09cb987b2ceb/template/android/gradle.properties#L30-L33) of your project:
+
+```
+# Use this property to specify which architecture you want to build.
+# You can also override it from the CLI using
+# ./gradlew <task> -PreactNativeArchitectures=x86_64
+reactNativeArchitectures=armeabi-v7a,arm64-v8a,x86,x86_64
+```
+
+Once you build a **release version** of your app, don't forget to remove those flags as you want to build an apk/app bundle that works for all the ABIs and not only for the one you're using in your daily development workflow.
+
+## Use a compiler cache
```

Comment:
```suggestion
## C++
```
nit, for consistency with the other two sections, I'd have the top title be the tech area the suggestion is for.

Maybe at this point it'd be worth have this levelling
```
## C++
### Use a compiler cache
#### Local caches
#### Distributed caches
```

Reply:
(in doing so, it'd be more consistent with the iOS section too)

Reply:
I don't agree with this one 🤔 The `Build only one ABI` section is actually impacting the C++ build. Having a C++ section would convey the meaning that the sections are mutually exclusive.

Reply:
makes sense in the context of your reply to the other section of the existing section being Android specific - will resolve this